packages/editor/src/components/global-styles-provider/index.js

```diff
@@ -146,13 +146,17 @@ function useGlobalStylesUserConfig() {
 
 function useGlobalStylesBaseConfig() {
 	const baseConfig = useSelect( ( select ) => {
-		const { __experimentalGetCurrentThemeBaseGlobalStyles, canUser } =
-			select( coreStore );
-
-		return (
-			canUser( 'read', { kind: 'root', name: 'theme' } ) &&
-			__experimentalGetCurrentThemeBaseGlobalStyles()
-		);
+		const {
+			__experimentalGetCurrentThemeBaseGlobalStyles,
+			getCurrentTheme,
+			canUser,
+		} = select( coreStore );
+		const currentTheme = getCurrentTheme();
+
+		return currentTheme &&
+			canUser( 'read', 'global-styles/themes', currentTheme.stylesheet )
```

Comment:
This would match the request made by `__experimentalGetCurrentThemeBaseGlobalStyles`.

https://github.com/WordPress/gutenberg/blob/537fb1800ce2784388ec3d130a37bd26f6f2f62f/packages/core-data/src/resolvers.js#L648